src/renderer/initializers/workload-events.tsx

```diff
@@ -0,0 +1,38 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import { withInjectables } from "@ogre-tools/injectable-react";
+import { observer } from "mobx-react";
+import React from "react";
+import type { IsAllowedResource } from "../../common/utils/is-allowed-resource.injectable";
+import isAllowedResourceInjectable from "../../common/utils/is-allowed-resource.injectable";
+import { Events } from "../components/+events/events";
+
+export interface WorkloadEventsProps {}
+
+interface Dependencies {
+  isAllowedResource: IsAllowedResource;
+}
+
+const NonInjectedWorkloadEvents = observer(({ isAllowedResource }: Dependencies & WorkloadEventsProps) => {
+  if (!isAllowedResource("events")) {
```

Comment:
Note: This type of defensive code should stay outside of component. (e.g. where it is used). Invites complicated logic inside a component which is not welcome. Future improvement.